test/functional/rpc_psbt.py

```diff
@@ -210,6 +210,55 @@ def assert_change_type(self, psbtx, expected_type):
         assert_equal(decoded_psbt["tx"]["vout"][changepos]["scriptPubKey"]["type"], expected_type)
 
     def run_test(self):
+
+        self.log.info("Test that PSBT can have user-provided UTXOs filled and signed")
+
+        # Create 1 parent 1 child chain from same wallet
+        psbtx_parent = self.nodes[0].walletcreatefundedpsbt([], {self.nodes[0].getnewaddress():10})['psbt']
+        processed_parent = self.nodes[0].walletprocesspsbt(psbtx_parent)
+        parent_txinfo = self.nodes[0].decoderawtransaction(processed_parent["hex"])
+        parent_txid = parent_txinfo["txid"]
+        parent_vout = 0 # just take the first output to spend
+
+        psbtx_child = self.nodes[0].createpsbt([{"txid": parent_txid, "vout": parent_vout}], {self.nodes[0].getnewaddress(): parent_txinfo["vout"][0]["value"] - Decimal("0.01")})
+
+        # Can not sign due to lack of utxo
+        res = self.nodes[0].walletprocesspsbt(psbtx_child)
+        assert not res["complete"]
```

Comment:
Outside the scope of this PR: While testing I realised that the output does not display the reason why it failed, would it be a good idea to display the error as well?
```
➜  src git:(2024-09-updateutxo_psbt) ✗ bitcoincli walletprocesspsbt cHNidP8BAFICAAAAAfDLyQnCSPG31wZz1rHABK8QVnQLIubUuakp3tA1b688AAAAAAD9////AQBlzR0AAAAAFgAU0NrK8DrQVPpN/ZzmgNtNu55HmNgAAAAAAAAA
{
  "psbt": "cHNidP8BAFICAAAAAfDLyQnCSPG31wZz1rHABK8QVnQLIubUuakp3tA1b688AAAAAAD9////AQBlzR0AAAAAFgAU0NrK8DrQVPpN/ZzmgNtNu55HmNgAAAAAAAAiAgNKBeuPVHjpCVZn1sLhqf9axYBqutVFf3QOBrh0J+qI4hi/EclnVAAAgAEAAIAAAACAAAAAAAEAAAAA",
  "complete": false
}
```